lib/matplotlib/units.py

```diff
@@ -157,11 +157,15 @@ def get_converter(self, x):
                     converter = self.get_converter(next_item)
                 return converter
 
-        if converter is None and iterable(x) and (len(x) > 0):
-            thisx = safe_first_element(x)
-            if classx and classx != getattr(thisx, '__class__', None):
-                converter = self.get_converter(thisx)
-                return converter
+        if converter is None:
+            try:
+                thisx = safe_first_element(x)
+            except (TypeError, StopIteration):
+                pass
+            else:
+                if classx and classx != getattr(thisx, '__class__', None):
+                    converter = self.get_converter(thisx)
+                    return converter
```

Comment:
the converter is always returned at the bottom of this function (it's literally the next line) so is it worth putting here? And a general question/qualm on the multiple returns: while I know PEP8 is lenient on them, is there a good reason for them here where everything is bundled in if statements anyway?